files/xml.py

```diff
@@ -0,0 +1,470 @@
+                if is_attribute(tree, xpath, namespaces):
+                    # Delete an attribute
+                    parent = result.getparent()
+                    # Pop this attribute match out of the parent
+                    # node's 'attrib' dict by using this match's
+                    # 'attrname' attribute for the key
+                    parent.attrib.pop(result.attrname)
+                elif is_node(tree, xpath, namespaces):
+                    # Delete an element
+                    result.getparent().remove(result)
+    except Exception, e:
+        abort(module, "Couldn't delete xpath target: %s (%s)" % (xpath, str(e)))
+    else:
+        finish(module, tree, xpath, namespaces, changed=True)
+
+def set_target_children(module, tree, xpath, namespaces, children):
+    matches = tree.xpath(xpath, namespaces=namespaces)
+
+    # Create a list of our new children
+    children = children_to_nodes(module=module, children=children)
+    children_as_string = [lxml.etree.tostring(c) for c in children]
+
+    changed = False
+
+    def replace_children_of(match):
+        if not module.check_mode:
+            for element in match.getchildren():
+                match.remove(element)
+            match.extend(children)
+
+    # xpaths always return matches as a list, so....
+    for match in matches:
+        # Check if elements differ
+        if len(match.getchildren()) == len(children):
+            for idx, element in enumerate(match.getchildren()):
+                if lxml.etree.tostring(element) != children_as_string[idx]:
+                    replace_children_of(match)
+                    changed = True
+                    break
+        else:
+            replace_children_of(match)
+            changed = True
+
+    # Write it out
+    finish(module, tree, xpath, namespaces, changed=changed)
+
+def add_target_children(module, tree, xpath, namespaces, children):
+    if is_node(tree, xpath, namespaces):
+        new_kids = children_to_nodes(module, children)
+        for node in tree.xpath(xpath, namespaces=namespaces):
+            if not module.check_mode: node.extend(new_kids)
+        finish(module, tree, xpath, namespaces, changed=True)
+    else:
+        finish(module, tree, xpath, namespaces)
+
+def set_target(module, tree, xpath, namespaces, attribute, value):
+    changed = False
+
+    if not is_node(tree, xpath, namespaces):
+        abort(module, "Xpath " + xpath + " does not reference a node!")
+
+    for element in tree.xpath(xpath, namespaces=namespaces):
+        if not attribute:
+            changed = changed or (element.text != value)
+            if not module.check_mode and (element.text != value): element.text = value
+        else:
+            changed = changed or (element.get(attribute) != value)
+            if ":" in attribute:
+                attr_ns, attr_name = attribute.split(":")
+                attribute = "{{{0}}}{1}".format(namespaces[attr_ns], attr_name)
```

Comment:
format is another python-2.6+ ism.  If there's a need to make this compatible with python-2.4 it will have to change to percent formatting instead.